src/containers/edm/propertytypes/PropertyTypeDetailsContainer.js

```diff
@@ -46,14 +49,74 @@ class PropertyTypeDetailsContainer extends React.Component<Props> {
     }
   }
 
+  renderEntityTypesSection = (entityTypes :List<Map<*, *>>) => {
+
+    if (entityTypes.isEmpty()) {
+      return null;
+    }
+
+    let entityTypesDataTable :React$Node = (
+      <AbstractTypeDataTable
+          abstractTypes={entityTypes}
+          maxHeight={500}
+          workingAbstractTypeType={AbstractTypes.EntityTypes} />
+    );
+
+    if (AuthUtils.isAuthenticated() && AuthUtils.isAdmin()) {
+      entityTypesDataTable = (
+        <AbstractTypeDataTable
+            abstractTypes={entityTypes}
+            highlightOnHover
+            maxHeight={500}
+            onAbstractTypeRemove={this.handleOnEntityTypeRemove}
+            onReorder={this.handleOnEntityTypeReorder}
+            orderable
+            showRemoveColumn
+            workingAbstractTypeType={AbstractTypes.EntityType} />
+      );
+    }
+
+    return (
+      <section>
+        <h2>EntityTypes</h2>
+        { entityTypesDataTable }
+      </section>
+    );
+  }
+
   render() {
 
     if (!this.props.propertyType || this.props.propertyType.isEmpty()) {
       return null;
     }
-
     const ptPII :boolean = this.props.propertyType.get('piiField', false);
     const piiAsString :string = ptPII === true ? 'true' : 'false';
+    // This gathers ALL entityTypes from the EDM
```

Comment:
1. Let's move all of this logic to `mapStateToProps`
2. You can simplify this significantly using `filter()`

https://facebook.github.io/immutable-js/docs/#/List/filter